src/WhatToCook.Application/Services/MealPlanningService.cs

```diff
@@ -0,0 +1,66 @@
+﻿using WhatToCook.Application.DataTransferObjects.Requests;
+using WhatToCook.Application.Domain;
+using WhatToCook.Application.Infrastructure.Repositories;
+
+namespace WhatToCook.Application.Services;
+
+public class MealPlanningService
+{
+    private readonly IRecipesRepository _recipesRepository;
+    private readonly IMealPlanningRepository _mealPlanningRepository;
+
+    public MealPlanningService(IRecipesRepository recipesRepository, IMealPlanningRepository mealPlanningRepository)
+    {
+        _recipesRepository = recipesRepository;
+        _mealPlanningRepository = mealPlanningRepository;
+    }
+
+    public async Task<PlanOfMeals> Create(PlanOfMealRequest planOfMealRequest)
+    {
+        var getRecipeForMealPlan = planOfMealRequest.Recipes.Select(x => x.Name);
+        var recipes = _recipesRepository.GetByNames(getRecipeForMealPlan);
+
+        var planOfMeals = new PlanOfMeals()
+        {
+            Name = planOfMealRequest.Name,
+            Id = planOfMealRequest.Id,
+            FromDate = DateTime.SpecifyKind(planOfMealRequest.FromDate, DateTimeKind.Utc),
+            ToDate = DateTime.SpecifyKind(planOfMealRequest.ToDate, DateTimeKind.Utc),
+            Recipes = recipes
+        };
+
+        await _mealPlanningRepository.Create(planOfMeals);
+
+        return planOfMeals;
+    }
+
+    public async Task<PlanOfMeals> Update(UpdatePlanOfMealRequest planOfMealRequest)
+    {
+        var getMealPlanToUpdate = await _mealPlanningRepository.GetMealPlanByName(planOfMealRequest.Name);
+        var getRecipeForMealPlanUpdate = planOfMealRequest.Recipes.Select(x => x.Name);
+
+        var recipes = _recipesRepository.GetByNames(getRecipeForMealPlanUpdate);
+        if (getMealPlanToUpdate == null)
+        {
+            throw new Exception($"Cannot Update {planOfMealRequest.Name}");
```

Comment:
Maybe let's create some custom exception for this kind of stuff? Maybe NotFoundException?
https://learn.microsoft.com/en-us/dotnet/standard/exceptions/how-to-create-user-defined-exceptions

Reply:
done, will be in the next commit